Image Contrast Adjusting Filter/Image Contrast Adjusting Filter.py

```diff
@@ -0,0 +1,82 @@
+
+from IPython.display import display, Math, Latex
+
+import numpy as np
+import matplotlib.pyplot as plt
+from PIL import Image
+
+img_path = input("Enter the path here:")  #example -> C:\Users\xyz\OneDrive\Desktop\project\image.jpg 
+
+img = Image.open(img_path)
+
+# convert our image into a numpy array
+img = np.asarray(img)
+#print(img.shape)
+# put pixels in a 1D array by flattening out img array
+flat = img.flatten()
+
+# show the histogram
+
+
+# create our own histogram function
+def get_histogram(image, bins):
+    # array with size of bins, set to zeros
+    histogram = np.zeros(bins)
+    
+    # loop through pixels and sum up counts of pixels
+    for pixel in image:
+        histogram[pixel] += 1
+    
+    # return our final result
+    return histogram
+
+# execute our histogram function
+hist = get_histogram(flat, 256)
+
+# create our cumulative sum function
+def cumsum(a):
+    a = iter(a)
+    b = [next(a)]
+    for i in a:
+        b.append(b[-1] + i)
+    return np.array(b)
```

Comment:
you can use np.cumsum() function instead of this

Reply:
done!